recipes/pcre2/all/conanfile.py

```diff
@@ -41,33 +55,45 @@ def configure(self):
         del self.settings.compiler.cppstd
 
     def requirements(self):
+        self.requires("zlib/1.2.11")
         if self.options.with_bzip2:
             self.requires("bzip2/1.0.8")
 
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["PCRE2_BUILD_TESTS"] = False
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["PCRE2_BUILD_TESTS"] = False
         if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
             runtime = not self.options.shared and "MT" in self.settings.compiler.runtime
-            cmake.definitions["PCRE2_STATIC_RUNTIME"] = runtime
-        cmake.definitions["PCRE2_DEBUG"] = self.settings.build_type == "Debug"
-        cmake.definitions["PCRE2_BUILD_PCRE2_8"] = self.options.build_pcre2_8
-        cmake.definitions["PCRE2_BUILD_PCRE2_16"] = self.options.build_pcre2_16
-        cmake.definitions["PCRE2_BUILD_PCRE2_32"] = self.options.build_pcre2_32
-        cmake.definitions["PCRE2_SUPPORT_JIT"] = self.options.support_jit
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+            self._cmake.definitions["PCRE2_STATIC_RUNTIME"] = runtime
+        self._cmake.definitions["PCRE2_DEBUG"] = self.settings.build_type == "Debug"
+        self._cmake.definitions["PCRE2_BUILD_PCRE2_8"] = self.options.build_pcre2_8
+        self._cmake.definitions["PCRE2_BUILD_PCRE2_16"] = self.options.build_pcre2_16
+        self._cmake.definitions["PCRE2_BUILD_PCRE2_32"] = self.options.build_pcre2_32
+        self._cmake.definitions["PCRE2_SUPPORT_JIT"] = self.options.support_jit
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def patch_cmake(self):
+        """Patch CMake file to avoid man and share during install stage"""
+
+        cmake_file = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(cmake_file, "INSTALL(FILES ${man1} DESTINATION man/man1)", "")
+        tools.replace_in_file(cmake_file, "INSTALL(FILES ${man3} DESTINATION man/man3)", "")
+        tools.replace_in_file(cmake_file, "INSTALL(FILES ${html} DESTINATION share/doc/pcre2/html)", "")
 
```

Comment:
```suggestion
```

Keep the old behavior, it's not a problem removing after installing.

Reply:
its copy paste from the pcre recipe, Ill fix it